src/libraries/Microsoft.PowerFx.Core/Parser/ParserOptions.cs

```diff
@@ -44,16 +44,22 @@ public class ParserOptions
         /// </summary>
         public int MaxExpressionLength { get; set; }
 
-        public ParserOptions()
-        {
-        }
-
-        public ParserOptions(CultureInfo culture, bool allowsSideEffects = false, int maxExpressionLength = 0)
-        {
-            Culture = culture;
-            AllowsSideEffects = allowsSideEffects;
-            MaxExpressionLength = maxExpressionLength;
-        }
+        /// <summary>
+        /// Flag for parse type literals.
+        /// </summary>
+        internal bool AllowParseAsTypeLiteral { get; set; }
+
+        public ParserOptions()
+        {
+        }
+
+        public ParserOptions(CultureInfo culture, bool allowsSideEffects = false, int maxExpressionLength = 0, bool allowParseAsTypeLiteral = false)
+        {
+            Culture = culture;
+            AllowsSideEffects = allowsSideEffects;
+            MaxExpressionLength = maxExpressionLength;
+            AllowParseAsTypeLiteral = allowParseAsTypeLiteral;
```

Comment:
>AllowParseAsTypeLiteral [](http://example.com/codeflow?start=12&length=23)

Does this need to be part of ctor? I don't see other flags are set this way. Default value of bool is false anyway, I think.